syntax = "proto3";

package iroha.protocol;

import "block.proto";
import "queries.proto";
import "google/protobuf/empty.proto";
import "responses.proto";


enum TxStatus {
  STATELESS_VALIDATION_FAILED = 0;
  STATELESS_VALIDATION_SUCCESS = 1;
  STATEFUL_VALIDATION_FAILED = 2;
  STATEFUL_VALIDATION_SUCCESS = 3;
  COMMITTED = 4;
  IN_PROGRESS = 5;
  NOT_RECEIVED = 6;
}

message ToriiResponse {
  TxStatus tx_status = 1;
  bytes tx_hash = 2;
}

message TxStatusRequest{
  bytes tx_hash = 1;
}

service CommandService {
  rpc Torii (Transaction) returns (google.protobuf.Empty);
  rpc Status (TxStatusRequest) returns (ToriiResponse);
}


service QueryService {
  rpc Find (Query) returns (QueryResponse);
}

enum GenesisBlockApplied {
  APPLY_FAILURE = 0;
  APPLY_SUCCESS = 1;
}

message ApplyGenesisBlockResponse {
  GenesisBlockApplied applied = 1;
}

service GenesisBlockService {
  rpc SendGenesisBlock (Block) returns (ApplyGenesisBlockResponse);
  rpc SendAbortGenesisBlock (Block) returns (google.protobuf.Empty);
}
